Automattic/wp-calypso
# Store: Refactor product list fetch for params.

Instead of just allowing a page number, the product list fetch now
allows other parameters. This will be required to fetch products for the
promotions page. 

To Test:
1. `npm test`
2. Visit store product list page
3. Ensure product list and pages work.
4. Ensure search function and pages work.
